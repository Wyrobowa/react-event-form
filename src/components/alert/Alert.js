import React from 'react';
import PropTypes from 'prop-types';

const Alert = ({ type, msg, children }) => (
  <div className={`alert alert-${type} mt-3`}>
    <h4 className="alert-heading">{msg}</h4>
    {children}
  </div>
);

Alert.propTypes = {
  type: PropTypes.oneOf(['primary', 'secondary', 'success', 'danger', 'warning', 'info']).isRequired,
  msg: PropTypes.string.isRequired,
  children: PropTypes.node,
};

Alert.defaultProps = {
  children: '',
};

export default Alert;
