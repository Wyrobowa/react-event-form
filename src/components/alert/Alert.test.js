import React from 'react';
import { shallow } from 'enzyme';
import Alert from './Alert';

describe('Alert', () => {
  it('renders Alert component without crashing', () => {
    const component = shallow(<Alert type="primary" msg="Alert message" />);

    expect(component).toMatchSnapshot();
  });
});
