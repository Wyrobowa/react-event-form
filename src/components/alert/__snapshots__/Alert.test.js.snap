// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders Alert component without crashing 1`] = `
<div
  className="alert alert-primary mt-3"
>
  <h4
    className="alert-heading"
  >
    Alert message
  </h4>
</div>
`;
